internals/src/macros.rs

```diff
@@ -0,0 +1,63 @@
+//! Various macros used by the Rust Bitcoin ecosystem.
+//!
+
+/// Implements standard array methods for a given wrapper type
+#[macro_export]
+macro_rules! impl_array_newtype {
+    ($thing:ident, $ty:ty, $len:literal) => {
+        impl $thing {
+            /// Converts the object to a raw pointer
+            #[inline]
+            pub fn as_ptr(&self) -> *const $ty {
```

Comment:
I just realized this is effectively `unsafe` code. We don't explicitly say that `unsafe` code can rely on the validity of the returned pointer so it's till `unsafe` consumer to blame if we get it wrong but without guaranteeing validity the method is near useless - you can at best use it to print out the pointer. And `as_mut_ptr` is thus redundant.